packages/core/upload/admin/src/utils/urlsToAssets.js

```diff
@@ -2,6 +2,21 @@ import axios from 'axios';
 import { AssetSource } from '../constants';
 import { typeFromMime } from './typeFromMime';
 
+/**
+ * The URL might have query parameters.
+ * Remove them to have a clean URL instead to use as a filename.
+ */
+function cleanUrl(url) {
+  // Break URL at ? and take first part (file name, extension)
+  url = url.split('?')[0];
+
+  // Sometimes URL doesn't have ? but #
+  url = url.split('#')[0];
+
+  // Cleaned URL
+  return url;
+}
```

Comment:
i'd suggest to use the URL API instead: https://developer.mozilla.org/en-US/docs/Web/API/URL

```suggestion
function cleanUrl(url) {
  return new URL(url).pathname.substring(1);
}
```

Reply:
Oh great! did not know about the URL API. I will use that!

About handling it in the backend, we could think of a way to do it. 
We need to have in mind that we handle URL's and also normal file names. If we apply a URL filter in the backend, we might break it for some filenames like `myfile%#.jpg`.

What the backend API is requesting is a valid filename and not an URL, and because of that, I think it would be difficult to try to use the URL filtering in the backend.